test/e2e/e2ev2.go

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"sigs.k8s.io/kubebuilder/test/e2e/scaffold"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("kubebuilder", func() {
+	Context("with v2 scaffolding", func() {
+		imageName := "controller:v2.0.1"
+		var testSuffix string
+		var c *config
+		var kbTest *kubebuilderTest
+
+		BeforeEach(func() {
+			var err error
+			testSuffix, err = randomSuffix()
+			Expect(err).NotTo(HaveOccurred())
+			c, err = configWithSuffix(testSuffix)
+			Expect(err).NotTo(HaveOccurred())
+			kbTest = &kubebuilderTest{
+				Namespace: fmt.Sprintf("e2e-%s-system", testSuffix),
+				Dir:       c.workDir,
+				Env:       []string{"GO111MODULE=on"},
+			}
+			prepare(c.workDir)
+		})
+
+		AfterEach(func() {
+			By("clean up created API objects during test process")
+			resources, err := kbTest.RunKustomizeCommand("build", filepath.Join("config", "default"))
+			if err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when running kustomize build during cleaning up: %v\n", err)
+			}
+			if _, err = kbTest.RunKubectlCommandWithInput(resources, "delete", "--recursive", "-f", "-"); err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when running kubectl delete during cleaning up api objects: %v\n", err)
+			}
```

Comment:
We can probably combine these two commands ? `kustomize build config/default | kubectl delete -f - ` ?